sdk/core/core-amqp/src/errors.ts

```diff
@@ -539,6 +549,29 @@ function isBrowserWebsocketError(err: any): boolean {
   return result;
 }
 
+/**
+ * @internal
+ * Checks if given object maps to a valid custom error. If yes, configures and returns the appropriate error instance, else returns `undefined`.
+ * @param err
+ */
+function getCustomError(err: AmqpError | Error): MessagingError | undefined {
```

Comment:
We don't want to cover any of the amqp errors here because amqp errors should undergo the appropriate processing as covered in the `translate()`. Therefore, make an early exit from here if `isAmqpError(err)` returns `true`

Reply:
I see that we have updated the if/else in translate() to use the result of this method only when it is not an amqp error. That is good.
But here since the input supports both AmqpError and simple Error, it is still confusing to anyone reading this code independently

My first suggestion would have been to update this method to take in only `Error`, but this might not work out due to typing issues. Because as far as the typescript compiler is concerned, the `err` object you pass in could be either of the 2 types.

Please consider not having this as a separate method and moving the code here directly into `translate()`